x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AggregateMapper.java

```diff
@@ -131,50 +137,62 @@ private List<IntermediateStateDesc> getNonNull(AggDef aggDef) {
         return l;
     }
 
-    static Stream<Tuple<Class<?>, String>> typeAndNames(Class<?> clazz) {
+    private static Stream<Tuple<Class<?>, Tuple<String, String>>> typeAndNames(Class<?> clazz) {
         List<String> types;
+        List<String> extraConfigs = List.of("");
         if (NumericAggregate.class.isAssignableFrom(clazz)) {
             types = NUMERIC;
         } else if (clazz == Count.class) {
             types = List.of(""); // no extra type distinction
+        } else if (clazz == SpatialCentroid.class) {
+            types = SPATIAL;
+            extraConfigs = List.of("SourceValues", "DocValues");
```

Comment:
This is where the additional combination of function names is generated. So far only spatial aggregations need this, but the idea can be used by others if there is ever value.

Reply:
Could you explain what this is about in more detail? I'd kind of imagined the `withForStats` would be enough to enable the optimization and the mapper would just call the single function. I imagine I'm missing some part of how the linkage works though.

Reply:
Take a look at `SpatialCentroid.supplier()`. There are four possible combinations, based on two spatial types `GEO_POINT` and `CARTESIAN_POINT`, and two loading types `DocValues` and `SourceValues`. The code generator needs to generate functions for all four combinations. The `AggregateMapper` deals with many possible combinations, but not this specific set of `DocValues` and `SourceValues`, so I added these using the same pattern I see here for other combinations. Without this fix the classloader cannot find the classes to load for the four combinations.

Reply:
> Without this fix the classloader cannot find the classes to load for the four combinations.

That's the trick. I suppose we're using some kind of trick to get these linked up. Scalars just use booleans for it. I'd kind of prefer that. But now isn't the time to rework that.